torch/csrc/autograd/python_variable_indexing.h

```diff
@@ -1,11 +1,91 @@
 #pragma once
 
+#include <Exceptions.h>
+#include <c10/core/SymInt.h>
+#include <pyport.h>
+#include <sys/types.h>
 #include <torch/csrc/autograd/python_variable.h>
 #include <torch/csrc/python_headers.h>
+#include <utils/python_symnode.h>
 
 namespace torch {
 namespace autograd {
 
+struct UnpackedSlice {
+  c10::SymInt start;
+  c10::SymInt stop;
+  c10::SymInt step;
+};
+
+// This mirrors Cpython's PySlice_Unpack method
+inline UnpackedSlice __PySlice_Unpack(PyObject* _r) {
```

Comment:
Is inlining this function intentional? I think it makes more sense to put definition into cpp file.